### PR TITLE
#162163003 Build out REST endpoints for Article  

### PR DESCRIPTION
**Pull request by katunold**
#### What does this PR do?
* Implement **CRUD** on article endpoint in ah REST API
* Write tests for the article app
#### Description of Task to be completed?
*Have the following endpoints working*
1. List Articles
`GET /api/articles/` To return a list of all articles
2. Get specific Article
`GET /api/articles/slug` To return a specific article basing on the slug provided
3. Create Article
`POST /api/articles/` To create and return the article created
Example request body
```
{
  "article": {
    "title": "Who is he?",
    "description": "He has a bald head",
    "body": "It is believed he will rule Uganda to heaven"
  }
}
```
4. Update Article
`PUT /api/articles/slug/` To make changes to any fields that are input by the user while creating the article. that is if the user posted the article
```
{
  "article": {
    "title": "Who is he? a god or demigod?"
  }
}
```
5. Delete Article
`DELETE /api/articles/slug` To delete an article only posted by the user
#### How should this be manually tested?
After cloning the repo, cd into it and follow the steps 
* In terminal, type `git checkout ft-create-article-162163003` to switch to my branch
* Open your prefered editor and go to the settings. py file in the project
* In the settings.py file, edit the value assigned to a password in Database to what you use locally on your computer
* Run the server using python manage.py runserver
* Using postman test every endpoint above using these headers:
_key_: content-type _value_: application/json
_key_: Authorization _value_: Token **your token**
#### What are the relevant pivotal tracker stories?
[#162163003](https://www.pivotaltracker.com/story/show/162163003)

### CODE REVIEW

**Review by aayebare — APPROVED** [review]
up and onwards. Great work

File: .travis.yml [hunk 1]
```diff
@@ -6,8 +6,10 @@ services:
...
 - pip3 install -r requirements.txt
 - psql -c 'create database ahdb;' -U postgres
+- python manage.py makemigrations
 - python manage.py migrate
 script:
-- coverage run --source='authors/apps' manage.py test && coverage report
+- coverage run --source='authors' manage.py test
+- coverage report
 after_success:
 - coveralls
```
Comment (malep2007) [inline .travis.yml:15]: New Line issues

File: authors/apps/article/models.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+from authors.apps.article.utils import generate_slug
+from authors.apps.authentication.models import User
+
+
+class Article(models.Model):
+
+    # Model for an article
+    author = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="articles", null=True)
```
Comment (malep2007) [inline authors/apps/article/models.py:17]: improve readability
```python
author = models.ForeignKey(
    User,
    on_delete = models.CASCADE,
    related_name = "articles",
    null=True
)
```

Comment (malep2007) [reply authors/apps/article/models.py:17]: Improve it throughout the codebase.

File: authors/apps/article/models.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    title = models.CharField(max_length=255, null=False, blank=False,
+                             error_messages={"required": "Add a title for your article."})
+    description = models.TextField(null=False, blank=False,
+                                   error_messages={"required": "Add a description for your article."})
```
Comment (malep2007) [inline authors/apps/article/models.py:21]: Even for structured dictionaries, improve the readability
```python
{
    "required": "Add a description for your article."
}
```

**Review by malep2007 — CHANGES_REQUESTED** [review]
Many issues with this PR, please address all concerns raised before you raise another PR.

File: .travis.yml (hunk 1 above)
Comment (katunold) [reply .travis.yml:15]: noted

File: authors/apps/article/models.py (hunk 3 above)
Comment (katunold) [reply authors/apps/article/models.py:21]: noted

**Pull request closed without merging**